Removing cluster theory prediction from Firecrown

eduardojsbarroso opened the pull request:
## Description

We are removing the cluster theory module from Firecrown, as it is now an independent library ([Crow](https://github.com/LSSTDESC/crow)). 

Fixes #580 

## Type of change

* Refactoring
* This change requires a documentation update

## Checklist:

The following checklist will make sure that you are following the code style and
guidelines of the project as described in the
[contributing](https://firecrown.readthedocs.io/en/latest/contrib.html) page.

- [x] I have run `bash pre-commit-check` and fixed any issues
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have made corresponding changes to the documentation
- [ ] I have 100% test coverage for my changes (please check this after the CI system has verified the coverage)

m-aguena committed:
replace files with crow

eduardojsbarroso committed:
working cosmosis with cluster_redshift_richness_deltasigma.py

vitenti committed:
Merge branch 'master' into issues/580/crow

eduardojsbarroso committed:
Merge branch 'master' into issues/580/crow

eduardojsbarroso committed:
Fixing imports, working example of cluster_redshift_richness_deltasigma.py

eduardojsbarroso committed:
Setup before calling recipe

eduardojsbarroso committed:
Implemented reduced shear data reading and examples

eduardojsbarroso committed:
Working examples for all cases

eduardojsbarroso committed:
Fixed reduced shear examples and black and pylint

eduardojsbarroso committed:
Fix tests, mypy and pylint

eduardojsbarroso committed:
Adding crow as dependency

eduardojsbarroso committed:
Trigger CI

eduardojsbarroso committed:
Changed environment

eduardojsbarroso committed:
Trigger CI

eduardojsbarroso committed:
Fixed parameter naming to for numcosmo run

eduardojsbarroso committed:
Fixed parameter name in test

eduardojsbarroso committed:
Changed objects to handle only shear or only deltasigma

marcpaterno committed:
Merge branch 'master' into issues/580/crow

eduardojsbarroso committed:
Removed wrong import from test

eduardojsbarroso committed:
Removed old tests

vitenti committed:
Merge branch 'master' into issues/580/crow

vitenti commented:
Hello @m-aguena and @eduardojsbarroso, we are reviewing your PR and would like to propose improving the interface between Crow and Firecrown. At the moment, we cannot run static analysis because Crow does not use type hints. We also see some Crow-specific code inside Firecrown that seems unnecessary.

A cleaner approach would be to restrict the interface to configuring Crow objects using the SACC data layout, then querying Crow for the relevant parameters and passing those values during the statistical analysis. To make this workable, it would help if you could add type hints to the interfaces used by Firecrown. With that in place, we can define a small protocol that Crow can implement, allowing us to rely on static analysis without making Crow depend on Firecrown.

We can prepare a minimal example and discuss the details in a meeting. What do you think?

marcpaterno committed:
Merge branch 'master' into issues/580/crow

eduardojsbarroso committed:
Fixed flake8 problems

eduardojsbarroso committed:
Black

eduardojsbarroso committed:
Fix bug with black and flake8

eduardojsbarroso committed:
Typo

eduardojsbarroso committed:
Fixed cyclic imports

eduardojsbarroso committed:
Fixed reset aprameters for very iteration

eduardojsbarroso committed:
Properly reset grid

eduardojsbarroso committed:
Changed reset line

eduardojsbarroso committed:
Added prior to prediction

eduardojsbarroso committed:
Fix typo

marcpaterno committed:
Correct behavior for CAMBExtraParams with different halofit versions.
Fixed minor linter problem in generate_sn_data.py.

marcpaterno committed:
Apply black

marcpaterno committed:
Fix imports and rename crow package

Rename the pip dependency from desc-crow to lsstdesc-crow in
environment.yml to match the updated package name.

Remove duplicate and stale imports from modeling_tools.py and
test_modeling_tools.py. The deprecated firecrown.ccl_factory import
path for CCLCalculatorArgs and CCLFactory was superseded by the
canonical module locations. ClusterAbundance and ClusterDeltaSigma
were also imported but unused.

marcpaterno commented:
We have merged the `master` branch into `issues/580/crow` to prepare for making a new release (firecrown 1.15) that includes a dependency on CROW. This mostly required updating imports to work with the new organization of the code. We did not modify any of the physics contents of the code.

But there are two problems to be fixed before we can merge the updated PR into master to make the new release of firecrown.

1. The installation of CROW is currently irregular.

We first tried `conda install lsstdecsc_crow`.
This obtained a package, but `import crow` does not work.
We then tried `pip install lsstdesc-crow`.
This installs version 1.0.4.
We would very much like to use the `conda` version, and to then update the `environment.yml` and the conda lockfiles to use that version.

2. The test coverage for firecrown.models.clusters is not at 100%. The CI is failing because of this.

If you pull the updates on this branch to your local clone and run `make test-models-clusters` you will see the record of untested lines. It appears that the `include_prop == ClusterProperty.SHEAR` branches are never exercised.

When these issues are fixed, we can merge the result into the `master` branch of firecrown and cut a new release (v1.15).

After that, we will pursue the refactoring that will more cleanly allow users of Firecrown to make use of multiple physics model libraries like CROW, without having Firecrown itself depend on any of the packages that define the models people use in their analysis code.

eduardojsbarroso committed:
Added tests for shear property in modeling

eduardojsbarroso committed:
Fixing integration tests

eduardojsbarroso committed:
Added final test for variance in binned_cluster

eduardojsbarroso committed:
Black
